package/MDAnalysis/coordinates/GRO.py

```diff
@@ -269,19 +275,37 @@ def write(self, selection, frame=None):
            resName and atomName are truncated to a maximum of 5 characters
         """
         # write() method that complies with the Trajectory API
-        u = selection.universe
-        if frame is not None:
-            u.trajectory[frame]  # advance to frame
-        else:
-            frame = u.trajectory.ts.frame
 
-        # make sure to use atoms (Issue 46)
-        atoms = selection.atoms
-        # can write from selection == Universe (Issue 49)
-        coordinates = atoms.positions
+        # if frame is not None:
+        #     u.trajectory[frame]  # advance to frame
+        # else:
+        #     frame = u.trajectory.ts.frame
+
+        try:
+
+            # make sure to use atoms (Issue 46)
+            atoms = obj.atoms
+            # can write from selection == Universe (Issue 49)
+
+        except AttributeError:
+            if isinstance(obj, base.Timestep):
+                selection = obj.copy()
+            else:
+                raise TypeError("No Timestep found in obj argument")
+        finally:
+            if isinstance(obj, AtomGroup):
+                selection = obj
+            elif hasattr(obj, 'universe'):
+                ts_full = obj.universe.trajectory.ts.copy()
+                if ts_full.n_atoms == atoms.n_atoms:
+                    selection = ts_full
+                else:
+                    selection = ts_full.copy_slice(atoms.indices)
```

Comment:
why not copy the timestep from the atoms object directly? There isn't a need for this if/else clause. 

Also as a side note. `selection` is not a useful name for the variable. By convention in the code we use the name selection for variables that contain selection strings. You should rather rename it to `timestep`/`ts`